### PR TITLE
FORMS-867: Export CSV submission columns issue (new validation for 'form' keyword in fieldnames)

### PR DESCRIPTION
**Pull request by bcvesalink**
# Description

This PR is to fix use cases when user would name one of the fields as `form` which is system keyword, so we added additional validation so users can't use this API name for fields

## Types of changes

Bug fix (non-breaking change which fixes an issue)

## Checklist

- [ ] I have read the [CONTRIBUTING](CONTRIBUTING.md) doc
- [ ] I have checked that unit tests pass locally with my changes
- [ ] I have run the npm script lint on the frontend and backend
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)
- [ ] I have approval from the product owner for the contribution in this pull request

### CODE REVIEW

File: app/frontend/src/internationalization/trans/chefs/en/en.json [hunk 1]
```diff
@@ -325,7 +325,8 @@
     "version": "Version",
     "save": "Save",
     "saving": "Saving",
-    "notSaved": "Not Saved"
+    "notSaved": "Not Saved",
+    "fieldname_error": "You CAN NOT use 'form' keyword as {label} fieldname"
```
Comment (WalterMoar) [inline app/frontend/src/internationalization/trans/chefs/en/en.json:329]: This is fairly minor, but so far we've never used an underscore in a translation key. For consistency can you change this to `fieldnameError`?

File: app/frontend/src/components/designer/FormDesigner.vue [hunk 2]
```diff
@@ -529,6 +532,18 @@ export default {
               this.addPatchToHistory();
             }
           }
+          this.canSave = true;
+          modified?.components?.map((comp) => {
+            if (comp.key === 'form') {
```
Comment (WalterMoar) [inline app/frontend/src/components/designer/FormDesigner.vue:537]: My concern here is that it's a breaking change. A form that was fine before this PR will produce an error when edited and saved after this PR, even if there are no changes to the form.

Maybe this isn't a big deal - how hard would it be to identify how many forms already in production use "form" as one of the field names? Maybe we could work with those form owners to fix their forms before the change.

**Review by WalterMoar — CHANGES_REQUESTED** [review]

File: app/frontend/src/internationalization/trans/chefs/en/en.json (hunk 1 above)
Comment (bcvesalink) [reply app/frontend/src/internationalization/trans/chefs/en/en.json]: @WalterMoar addressed

File: app/frontend/src/components/designer/FormDesigner.vue (hunk 2 above)
Comment (bcvesalink) [reply app/frontend/src/components/designer/FormDesigner.vue:537]: @WalterMoar there are only 2 published forms which have `form` as a field name, should we just notify them?

**Review by WalterMoar — APPROVED** [review]

**Pull request merged**